app/models/user_action.rb

```diff
@@ -411,7 +411,7 @@ def self.apply_common_filters(builder, user_id, guardian, ignore_private_message
       visible_post_types: visible_post_types,
     )
 
-    unless (guardian.user && guardian.user.id == user_id) || guardian.is_staff?
+    if !(guardian.user && guardian.user.id == user_id) && !guardian.is_staff?
```

Comment:
I _think_ that's the same thing but "easier" to read?

```suggestion
    if guardian.user&.id != user_id && !guardian.is_staff?
```